cmd/kubeadm/README.md

```diff
@@ -5,42 +5,63 @@
 ### `kubeadm init`
 
 It's usually enough to run `kubeadm init`, but in some case you might like to override the
-default behaviour.
+default behaviour. The flags used for said purpose are described below.
 
-- `--token=<str>`
+- `--token=<token>`
 
-By default, a token is generated, but if you are to automate cluster deployment, you want to
+By default, a token is generated, but if you are to automate cluster deployment, you will want to
 set the token ahead of time. Read the docs for more information on the token format.
 
-- `--api-advertise-addr=<ip>` (multiple values allowed)
-- `--api-external-dns-name=<domain>` (multiple values allowed)
+- `--api-advertise-addresses=<ips>` (multiple values are allowed by having multiple flag declarations or multiple values separated by comma)
+- `--api-external-dns-names=<domain>` (multiple values are allowed by having multiple flag declarations or multiple values separated by comma)
 
-By default, `kubeadm` will auto detect IP address and use that to generate API server certificates.
-If you would like to access the API via any external IPs and/or DNS, which it might not be able
-to detect, you can use `--api-advertise-addr` and `--api-external-dns-name` to add multiple
-different IP addresses and DNS names.
+By default, `kubeadm` will auto detect IP addresses and use that to generate API server certificates.
+If you would like to access the API via any external IPs and/or hostnames, which it might not be able
+to detect, you can use `--api-advertise-addresses` and `--api-external-dns-names` to add multiple
+different IP addresses and hostnames (DNS).
+
+- `--service-cidr=<cidr>` (default: "100.64.0.0/12")
+
+By default, `kubeadm` sets `100.64.0.0/12` as the subnet for services. This means when a service is created, its cluster IP, if not manually specified, 
+will be automatically assigned from the services subnet. If you would like to set a different one, use `--service-cidr`.
 
-- `--service-cidr=<cidr>` (default: "100.64/12")
 - `--service-dns-domain=<domain>` (default: "cluster.local")
 
-- `--use-hyperkube=<bool>` (default: "false")
+By default, `kubeadm` sets `cluster.local` as the cluster DNS domain. If you would like to set a different one, use `--service-dns-domain`.
+
+- `--schedule-workload=<bool>` (default: "false")
+
+By default, `kubeadm` sets the master node kubelet as non-schedulable for workloads. This means the master node won't run your pods. If you want to change that, 
+use `--schedule-workload=true`.
+
+- `--cloud-provider=<cloud provider>`
+
+By default, `kubeadm` doesn't perform auto-detection of the current cloud provider. If you want to specify it, use `--cloud-provider`. Possible values are
+the ones supported by controller-manager, namely `"aws"`, `"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`, `"rackspace"`, `"vsphere"`.
```

Comment:
What happens if I set this value? Do I need to do anything special with IAM roles? This needs to be explained here, or we need to link to some documentation on the matter.